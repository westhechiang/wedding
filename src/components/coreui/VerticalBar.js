import PropTypes from 'prop-types';
import styled from 'styled-components';
import React from 'react';
import { theme } from '../../ui';

const VerticalBar = ({ className }) => <span className={className}>|</span>;

const sizes = {
  xsmall: theme.font.xsmall,
  xxsmall: theme.font.xxsmall,
};

const StyledVerticalBar = styled(VerticalBar)`
  color: ${theme.color.white};
  font-family: ${theme.font.family};
  font-size: ${props => sizes[props.size]};
`;

VerticalBar.propTypes = {
  className: PropTypes.string,
};

export default StyledVerticalBar;
