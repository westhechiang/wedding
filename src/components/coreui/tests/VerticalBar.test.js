import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import VerticalBar from '../../../components/coreui/VerticalBar';

describe('<VerticalBar />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<VerticalBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xsmall" size snapshot', () => {
    const wrapper = shallow(<VerticalBar size="xsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xxsmall" size snapshot', () => {
    const wrapper = shallow(<VerticalBar size="xxsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
