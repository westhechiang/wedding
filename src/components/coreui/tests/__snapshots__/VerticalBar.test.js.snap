// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalBar /> should match its "xsmall" size snapshot 1`] = `
<VerticalBar
  className="sc-bdVaJa hOaGXn"
  size="xsmall"
/>
`;

exports[`<VerticalBar /> should match its "xxsmall" size snapshot 1`] = `
<VerticalBar
  className="sc-bdVaJa lnKfDf"
  size="xxsmall"
/>
`;

exports[`<VerticalBar /> should match its empty snapshot 1`] = `
<VerticalBar
  className="sc-bdVaJa dtmAum"
/>
`;
